packages/documentation/src/stories/utilities/spacing/spacing.stories.ts

```diff
@@ -1,15 +1,165 @@
+  'huge',
+  'giant',
+  'bigger-giant',
+];
+
+const positionOptions = {
+  '': 'All around',
+  'x': 'Along the horizontal axis',
+  'y': 'Along the vertical axis',
+  't': 'At the top',
+  'b': 'At the bottom',
+  'r': 'To the right',
+  'l': 'To the left',
+};
 
 const meta: Meta = {
   title: 'Utilities/Spacing',
   parameters: {
-    badges: [BADGE.TODO],
+    badges: [BADGE.NEEDS_REVISION],
+  },
+  args: {
+    marginSize: 'regular',
+    marginPosition: '',
+    paddingSize: 'regular',
+    paddingPosition: '',
+  },
+  argTypes: {
+    marginSize: {
+      name: 'Margin size',
+      description: 'Sets the size of the Margin.',
+      control: {
+        type: 'select',
+      },
+      options: sizingOptions,
+      table: {
+        category: 'General',
+      },
+    },
+    marginPosition: {
+      name: 'Margin Position',
+      description: 'Sets the position of the Margin.',
+      control: {
+        type: 'select',
+        labels: positionOptions,
+      },
+      options: Object.keys(positionOptions),
+      table: {
+        category: 'General',
+      },
+    },
+    paddingSize: {
+      name: 'Padding size',
+      description: 'Sets the size of the Padding.',
+      control: {
+        type: 'select',
+      },
+      options: sizingOptions,
+      table: {
+        category: 'General',
+      },
+    },
+    paddingPosition: {
+      name: 'Pading Position',
+      description: 'Sets the position of the Padding.',
+      control: {
+        type: 'select',
+        labels: positionOptions,
+      },
+      options: Object.keys(positionOptions),
+      table: {
+        category: 'General',
+      },
+    },
   },
 };
 
 export default meta;
 
 type Story = StoryObj;
 
-export const Default: Story = {};
+export const Default: Story = {
+  render: (args: Args) => {
+    return html`
+      <!-- Demo -->
+      <div class="bg-primary p-regular d-flex align-items-start">
+        <div class="flex-fill d-flex">
+          <div class="bg-petrol">
+            <div
+              class="bg-petrol-bright border border-dark h-bigger-giant m${args.marginPosition}-${args.marginSize} p${args.paddingPosition}-${args.paddingSize} w-bigger-giant"
+            >
+              <div class="bg-light h-100"></div>
+            </div>
+          </div>
+        </div>
+
+        <!-- Legend -->
+        <ul class="ps-regular text-white">
+          <li class="d-flex align-items-center mb-regular">
+            <div class="bg-petrol h-regular w-regular me-mini"></div>
+            <span>margin</span>
+          </li>
+          <li class="d-flex align-items-center mb-regular">
+            <div class="bg-petrol-bright h-regular w-regular me-mini"></div>
+            <span>padding</span>
+          </li>
+          <li class="d-flex align-items-center mb-regular">
+            <div class="bg-light h-regular w-regular me-mini"></div>
+            <span>content</span>
+          </li>
+        </ul>
+      </div>
+    `;
+  },
+};
```

Comment:
It would be nice to hide some code from the example output.
This is, because I would move most of the content to a decorator.

Try out the example below:

Create a styles file (`spacing.styles.scss`)and import it in your spacing.stories.ts like so: `@import './spacing.styles.scss';`
```scss
@use '@swisspost/design-system-styles/core' as post;

.spacing-example {
  .margin {
    background-color: #f4cea3;
  }
  .padding {
    background-color: #c4dab9;
    border: 1px solid post.$gray-40;
  }
  .content {
    background-color: post.$white;
  }

  .legend {
    .padding {
      border: 0 none;
    }

    .content {
      border: 1px solid post.$gray-40;
    }
  }
} 
```

Change your default story to the following:
```ts
export const Default: Story = {
  render: (args: Args) => {
    // this will be the only code visible in the code preview
    return html`
      <div
        class="w-bigger-giant h-bigger-giant m${args.marginPosition}-${args.marginSize} p${args.paddingPosition}-${args.paddingSize}"
      ></div>
    `;
  },
  decorators: [
    // everything in here will be visible in the example, but only the content coming from the `story` function will be shown in the code preview
    (story: StoryFn, { args, context }: StoryContext) => html`
      <div class="spacing-example d-flex gap-3">
        <div class="d-none">${story(args, context)}</div>

        <div class="d-flex flex-fill">
          <div
            class="margin ${['', 'x', 's', 'e'].includes(args.marginPosition as string) &&
            args.marginSize === 'auto'
              ? 'w-100'
              : ''}"
          >
            <div
              class="w-bigger-giant h-bigger-giant padding m${args.marginPosition}-${args.marginSize} p${args.paddingPosition}-${args.paddingSize}"
            >
              <div class="h-100 content"></div>
            </div>
          </div>
        </div>

        <ul class="legend list-unstyled">
          <li class="d-flex align-items-center">
            <div class="h-regular w-regular me-mini margin"></div>
            <span>margin</span>
          </li>
          <li class="d-flex align-items-center">
            <div class="h-regular w-regular me-mini padding"></div>
            <span>padding</span>
          </li>
          <li class="d-flex align-items-center">
            <div class="h-regular w-regular me-mini content"></div>
            <span>content</span>
          </li>
        </ul>
      </div>
    `,
  ],
};

```